src/Api.php

```diff
@@ -0,0 +1,626 @@
+                        'contents' => file_get_contents($dataDir . $filename),
+                    ],
+                ]
+            );
+            $response = $this->client->request(
+                'post',
+                sprintf('%s/storage/fs/upload', $this->config->getUrlAddress()),
+                [
+                    'headers' => [
+                        'Authorization' => 'Bearer ' . $this->accessToken,
+                        'x-upload-token' => $this->validateFile($dataDir, $filename),
+                        'Content-Type' => 'multipart/form-data; boundary=' . $streamFile->getBoundary(),
+                    ],
+                    'body' => $streamFile,
+                ]
+            );
+        } catch (ClientException $exception) {
+            throw new UserException($exception->getMessage(), $exception->getCode(), $exception);
+        } catch (ServerException $exception) {
+            throw new UserException($exception->getMessage(), $exception->getCode(), $exception);
+        }
+
+        $responseJson = json_decode($response->getBody()->getContents(), true);
+        return $responseJson[0]['storageInfo']['path'];
+    }
+
+    public function checkDatamodelExists(string $datamodelName): ?array
+    {
+        try {
+            $response = $this->client->get(
+                sprintf('%s/api/v2/datamodels/schema', $this->config->getUrlAddress()),
+                [
+                    'headers' => [
+                        'Authorization' => 'Bearer ' . $this->accessToken,
+                    ],
+                    'query' => [
+                        'title' => $datamodelName,
+                    ],
+                ]
+            );
+        } catch (ServerException $exception) {
+            return null;
+        }
+
+        return json_decode($response->getBody()->getContents(), true);
+    }
+
+    public function createDatamodel(string $datamodelName): array
+    {
+        try {
+            $response = $this->client->post(
+                sprintf('%s/api/v2/datamodels', $this->config->getUrlAddress()),
+                [
+                    'headers' => [
+                        'Authorization' => 'Bearer ' . $this->accessToken,
+                    ],
+                    'json' => [
+                        'title' => $datamodelName,
+                    ],
+                ]
+            );
+        } catch (ClientException $exception) {
+            throw new UserException($exception->getMessage(), $exception->getCode(), $exception);
+        } catch (ServerException $exception) {
+            throw new UserException($exception->getMessage(), $exception->getCode(), $exception);
+        }
+
+        return json_decode($response->getBody()->getContents(), true);
+    }
+
+    public function deleteDatamodel(string $datamodelId): void
+    {
+        try {
+            $this->client->delete(
+                sprintf('%s/api/v2/datamodels/%s', $this->config->getUrlAddress(), $datamodelId),
```

Comment:
Vo vsetkych metodach okrem login si pridavas `baseUrl = $this->config->getUrlAddress()` a pridavas `Bearer` hlavicku. Mohlo by sa to hodit do privatnej metody ...  tam volat `$this->client->request($method, $uri, $options);` ... odstranilo by to duplicitný kod a teda aj zmensilo riziko preklepu/chyby.

Reply:
trochu jsem to přepsal a odstranil duplicitní kód